### PR TITLE
Add support hash router

### PR DESCRIPTION
**Pull request by lexich**
https://github.com/nanostores/router/issues/3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import { atom, onMount } from 'nanostores'
 
-export function createRouter(routes) {
+export function createRouter(routes, options) {
   let router = atom()
+  let isHashRouter = options && options.hash;
+
+  let locationPathname = isHashRouter
```
Comment (ai) [inline index.js:7]: Let’s rename variable

**Comment by ai** [comment]
I am not sure that this is a right API.

Often, hash navigation is independent of route’s navigation. For instance, I use hash navigation for popups, which can be open on every page.

In your suggestion I need to manually describe all possible hashes in URL schemas which is not useful.

**Comment by lexich** [comment]
> I am not sure that this is a right API.
> 
> Often, hash navigation is independent of route’s navigation. For instance, I use hash navigation for popups, which can be open on every page.
> 
> In your suggestion I need to manually describe all possible hashes in URL schemas which is not useful.

What API is preferable for you? Do you want to have the possibility of using both schemas?

**Comment by ai** [comment]
> What API is preferable for you? Do you want to have the possibility of using both schemas?

It is a question for you.

This task not just about writing a code, but about creating API (something like product manager work but for open-source library).

**Comment by mattmccray** [comment]
Well, this was exactly what I was looking for. 

When the application is the resource the URL is locating, it makes sense to use hash routing as pointers to internal views (not to be confused with state).

To get it to recognize routes I did need to change lines 7-9 to:

```javascript
  let getPathname = isHashRouter
    ? () => location.hash.replace(/^#/, '')
    : () => location.pathname
```

When it stripped out the leading slash, that lead to route recognization failure... Maybe because I defined my routes with the leading slash, now I think about it.

Anyway, thanks for this patch, it's been very useful for me!

**Comment by jordmantheman** [comment]
Another use case for hash routing is when your app is distributed via some object store like AWS S3. You can set your CDN to route `myhost.com -> s3://some-bucket/index.html` and SPA like routing works until you attempt to refresh, in which case the CDN will attempt to route `myhost.com/subpage -> s3://some-bucket/subpage` which results in a 404.

While there are more robust solutions to this problem like configuring the store to redirect 404s back to your index.html... hash routing is a simple solution as well.

Hash routing (`myhost.com/#/somesubpage`) prevents the hash request from hitting the server. React Router's hash router has been supporting this well.

While I'll admit I wouldn't want to use this solution for anything too serious, it's a quick and dirty way to host a webapp.

**Comment by ai** [comment]
0.16 was released with `router.get().hash`

**Pull request closed without merging**